File: CalculatorApp/CalculatorAppTests/CalculatorTests.m
#import "CalculatorTests.h"
#import "Calculator.h"

@implementation CalculatorTests

- (void)setUp {
    [super setUp];
    sut = [[Calculator alloc] init];
}

- (void)testAddMethod_zeroLengthInput_returnsZero {
    NSInteger expected = 0;
    NSInteger result = [sut add:@""];

    STAssertEquals(expected, result, @"Zero length input should return 0.");
}

- (void)testAddMethod_oneLengthInput_returnsEquivalent {
    NSInteger expected = 8;
    NSInteger result = [sut add:@"8"];

    STAssertEquals(expected, result, @"One length input should return equivalent.");
}

- (void)testAddMethod_twoLengthInput_returnsSum {
    NSInteger expected = 8;
    NSInteger result = [sut add:@"5,3"];

    STAssertEquals(expected, result, @"Two length input should return sum.");
}

- (void)testAddMethod_anyLengthInput_returnsSum {
    NSInteger howMany = arc4random() % 999;
    NSInteger expected = 0;
    NSMutableString *numbersToAdd = [NSMutableString string];

    for (NSInteger i = 0; i < howMany; i++) {
        [numbersToAdd appendString:[NSString stringWithFormat:@"%i,",i]];
        expected += i;
    }

    NSInteger result = [sut add:numbersToAdd];

    STAssertEquals(expected, result, @"Two length input should return sum.");
}

- (void)testAddMethod_newLineDelimiterInput_returnsSum {
    NSInteger expected = 12;
    NSInteger result = [sut add:@"5,3\n4"];

    STAssertEquals(expected, result, @"New line delimiter input should return sum.");
}

- (void)testAddMethod_duplicateDelimiterInput_throwsException {
    @try {
        [sut add:@"3,4\n,5"];
        STAssertTrue(false, @"Duplicate delimiters should throw exception");
    } @catch (NSException *ex) {
        STAssertEqualObjects(@"DuplicateDelimitersException", [ex name], @"Exception name should match expected");
        STAssertEqualObjects(@"You cannot input duplicate delimiters", [ex description], @"Exception description should match expected");
    }
}

- (void)testAddMethod_customDelimiterInput_returnsSum {
    NSInteger expected = 15;
    NSInteger result = [sut add:@"//@\n3@5,3\n4"];

    STAssertEquals(expected, result, @"Custom delimiter input should return sum.");
}


@end

File: CalculatorApp/CalculatorTests.py
from Calculator import Calculator

__author__ = 'dgadd'

import unittest


class MyTestCase(unittest.TestCase):
    def test_addMethod_zeroLengthInput_returnsZero(self):
        expected = 0
        sut = Calculator()
        result = sut.add("")
        self.assertEqual(expected, result, msg="Zero length input should return 0.")

    def test_addMethod_oneLengthInput_returnsEquivalent(self):
        expected = 8
        sut = Calculator()
        result = sut.add("8")
        self.assertEqual(expected, result, msg="One length input should return equivalent.")

    def test_addMethod_twoLengthInput_returnsSum(self):
        expected = 10
        sut = Calculator()
        result = sut.add("3,7")
        self.assertEqual(expected, result, msg="Two length input should return sum.")

if __name__ == '__main__':
    unittest.main()
